azkaban-common/src/main/java/azkaban/user/UserUtils.java

```diff
@@ -50,52 +50,63 @@ static void setupWatch(final Map<String, ParseConfigFile> configFileMap) {
     Preconditions.checkNotNull(configFileMap);
     Preconditions.checkArgument(configFileMap.size() > 0);
 
-    Runnable runnable = () -> {
-      WatchService watchService  = null;
-      // Map to store WatchKey to Dir mapping
-      final Map<WatchKey, Path> keys = new HashMap<>();
-      // A directory to config files multimap
-      final Multimap<Path, String> dirToFilesMap = HashMultimap.create();
-
-      // Iterate over each file.
-      for (Map.Entry<String, ParseConfigFile> entry : configFileMap.entrySet()) {
-        String fileName = entry.getKey();
-        ParseConfigFile parser = entry.getValue();
-        Preconditions.checkNotNull(fileName);
-        Preconditions.checkNotNull(parser);
-
-        final File file = new File(fileName);
-        if (!file.exists()) {
-          log.warn("Failed to reload config, user provided file " + fileName + " does not exist.");
-          continue;
-        }
+    final WatchService watchService;
+    try {
+      watchService = FileSystems.getDefault().newWatchService();
+    } catch (IOException e) {
+      log.warn(" Failed to create WatchService " + e.getMessage());
```

Comment:
should we rethrow the exception to notify caller the watchService is not setup correctly.

Reply:
okay.